src/ripple/app/ledger/impl/LedgerConsensusImp.cpp

```diff
@@ -0,0 +1,1876 @@
+            return;
+    }
+
+    uint256 signingHash;
+    auto v = std::make_shared<STValidation> (mPreviousLedger->getHash (),
+        consensus_.validationTimestamp (), mValPublic, false);
+    addLoad(v);
+    v->setTrusted ();
+    signingHash = v->sign (mValPrivate);
+        // FIXME: wrong supression
+    getApp().getHashRouter ().addSuppression (signingHash);
+    getApp().getValidations ().addValidation (v, "localMissing");
+    Blob validation = v->getSigned ();
+    protocol::TMValidation val;
+    val.set_validation (&validation[0], validation.size ());
+    consensus_.setLastValidation (v);
+    WriteLog (lsWARNING, LedgerConsensus) << "Sending partial validation";
+}
+
+void LedgerConsensusImp::beginAccept (bool synchronous)
+{
+    std::shared_ptr<SHAMap> consensusSet
+        = mAcquired[mOurPosition->getCurrentHash ()];
+
+    if (!consensusSet)
+    {
+        WriteLog (lsFATAL, LedgerConsensus)
+            << "We don't have a consensus set";
+        abort ();
+        return;
+    }
+
+    consensus_.newLCL (mPeerPositions.size (), mCurrentMSeconds, mNewLedgerHash);
+
+    if (synchronous)
+        accept (consensusSet);
+    else
+    {
+        getApp().getJobQueue().addJob (jtACCEPT, "acceptLedger",
+            std::bind (&LedgerConsensusImp::accept, shared_from_this (), consensusSet));
+    }
+}
+
+void LedgerConsensusImp::endConsensus ()
+{
+    getApp().getOPs ().endConsensus (mHaveCorrectLCL);
+}
+
+void LedgerConsensusImp::addLoad(STValidation::ref val)
+{
+    std::uint32_t fee = std::max(
+        getApp().getFeeTrack().getLocalFee(),
+        getApp().getFeeTrack().getClusterFee());
+    std::uint32_t ref = getApp().getFeeTrack().getLoadBase();
+    if (fee > ref)
+        val->setFieldU32(sfLoadFee, fee);
+}
+
+//------------------------------------------------------------------------------
+std::shared_ptr <LedgerConsensus>
+make_LedgerConsensus (ConsensusImp& consensus, int previousProposers,
+    int previousConvergeTime, InboundTransactions& inboundTransactions,
+    LocalTxs& localtx, LedgerHash const &prevLCLHash, Ledger::ref previousLedger,
+    std::uint32_t closeTime, FeeVote& feeVote)
+{
+    return std::make_shared <LedgerConsensusImp> (consensus, previousProposers,
+        previousConvergeTime, inboundTransactions, localtx, prevLCLHash,
+        previousLedger, closeTime, feeVote);
+}
+
+//------------------------------------------------------------------------------
+
+/** Apply a transaction to a ledger
+
+  @param engine       The transaction engine containing the ledger.
+  @param txn          The transaction to be applied to ledger.
+  @param openLedger   true if ledger is open
+  @param retryAssured true if the transaction should be retried on failure.
+  @return             One of resultSuccess, resultFail or resultRetry.
+*/
+static
+int applyTransaction (
+    TransactionEngine& engine,
+    STTx::ref txn,
+    bool openLedger,
+    bool retryAssured)
```

Comment:
I'm not a fan, and there's no point to javadoccing a static function (since its clearly not public)